From ebe3dce6b37afbe69e6de397258639ec89e9f784 Mon Sep 17 00:00:00 2001
From: Huang Rui <vowstar@gmail.com>
Date: Wed, 9 Feb 2022 12:44:26 +0800
Subject: [PATCH] [CORRECTIVE] .qmake.conf: fix python undefined reference

Code that embeds Python (rather than building an extension module)
needs to pass --embed to any python3-config --ldflags or --libs
invocation to build above Python 3.8.

More information:
https://docs.python.org/3.8/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build

Signed-off-by: Huang Rui <vowstar@gmail.com>
---
 .qmake.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.qmake.conf b/.qmake.conf
index 09ea20017..0329c2430 100644
--- a/.qmake.conf
+++ b/.qmake.conf
@@ -15,8 +15,8 @@ PYTHON_CONFIG=python3-config
 # if the compilation fails.
 # PYTHON_LIBS contains the library names and locations (-l and -L flags) needed in linking
 # with Python.
-PYTHON_C_FLAGS=$$system($$PYTHON_CONFIG --cflags | grep -o -e "-I[^[:space:]]*" | uniq)
-PYTHON_LIBS=$$system($$PYTHON_CONFIG --ldflags | grep -o -e "-[lL][^[:space:]]*")
+PYTHON_C_FLAGS=$$system($$PYTHON_CONFIG --cflags --embed | grep -o -e "-I[^[:space:]]*" | uniq)
+PYTHON_LIBS=$$system($$PYTHON_CONFIG --ldflags --embed | grep -o -e "-[lL][^[:space:]]*")
 
 
 #
